PagingWithNetworkSample/app/src/main/java/com/android/example/paging/pagingwithnetwork/reddit/ui/RedditActivity.kt

```diff
@@ -78,34 +80,38 @@ class RedditActivity : AppCompatActivity() {
         initSearch()
     }
 
+    override fun onDestroy() {
+        super.onDestroy()
+        refreshListener?.let { (list.adapter as? PostsAdapter)?.removeLoadStateListener(it) }
+        refreshListener = null
+    }
+
     private fun initAdapter() {
         val glide = GlideApp.with(this)
-        val adapter = PostsAdapter(glide) {
-            model.retry()
+        val adapter = PostsAdapter(glide)
+        list.adapter = adapter.withLoadStateHeaderAndFooter(
+                header = PostsLoadStateAdapter(adapter),
+                footer = PostsLoadStateAdapter(adapter)
+        )
+
+        refreshListener = { loadType: LoadType, loadState: LoadState ->
+            if (loadType == LoadType.REFRESH) {
+                swipe_refresh.isRefreshing = loadState == LoadState.Loading
+            }
+        }.also { refreshListener ->
```

Comment:
why this .also?

Reply:
The member is nullable, but moot since I realize I no longer need to unregister explicitly